internal/pkg/version/version.go

```diff
@@ -4,65 +4,110 @@
 package version
 
 import (
+	"bytes"
 	"fmt"
-	"strings"
+	"time"
 )
 
 var (
+	// BuildDate is the time of the git commit used to build the program,
+	// in RFC3339 format. It is filled in by the compiler via makefile.
+	BuildDate string
+
 	// GitCommit and GitDescribe are filled by the compiler using ldflags to
 	// provide useful Git information.
 	GitCommit   string
 	GitDescribe string
 
 	// Version is the semantic version number describing the current state of
-	// NOM.
+	// nomad-pack.
 	Version = "0.0.1"
 
 	// Prerelease designates whether the current version is within a prerelease
 	// phase. Typically, this will be "dev" to signify a development cycle or a
 	// release candidate phase such as alpha, beta.1, rc.1, or such.
-	Prerelease = "techpreview.4"
+	VersionPrerelease = "techpreview.4"
 
 	// Metadata allows us to provide additional metadata information to the
 	// version identifier. This is typically used to identify enterprise builds
 	// using the "ent" metadata string.
-	Metadata = ""
+	VersionMetadata = ""
 )
 
-// HumanVersion composes the parts of the version in a way that's suitable for
-// displaying to humans.
-func HumanVersion() string {
-	version := Version
-	release := Prerelease
+// VersionInfo
+type VersionInfo struct {
+	BuildDate         time.Time
+	Revision          string
+	Version           string
+	VersionPrerelease string
+	VersionMetadata   string
+}
+
+func (v *VersionInfo) Copy() *VersionInfo {
+	if v == nil {
+		return nil
+	}
+
+	nv := *v
+	return &nv
+}
 
+func GetVersion() *VersionInfo {
+	ver := Version
+	rel := VersionPrerelease
+	md := VersionMetadata
 	if GitDescribe != "" {
-		version = GitDescribe
-	} else {
-		if release == "" {
-			release = "dev"
-		}
-
-		if release != "" && !strings.HasSuffix(version, "-"+release) {
-			// if we tagged a prerelease version then the release is in the version
-			// already.
-			version += fmt.Sprintf("-%s", release)
-		}
-
-		if Metadata != "" {
-			version += fmt.Sprintf("+%s", Metadata)
-		}
+		ver = GitDescribe
+	}
+	if GitDescribe == "" && rel == "" && VersionPrerelease != "" {
+		rel = "dev"
+	}
+
+	// on parse error, will be zero value time.Time{}
+	built, _ := time.Parse(time.RFC3339, BuildDate)
+
+	return &VersionInfo{
+		BuildDate:         built,
+		Revision:          GitCommit,
+		Version:           ver,
+		VersionPrerelease: rel,
+		VersionMetadata:   md,
+	}
+}
+
+func (c *VersionInfo) VersionNumber() string {
+	version := c.Version
+
+	if c.VersionPrerelease != "" {
+		version = fmt.Sprintf("%s-%s", version, c.VersionPrerelease)
+	}
+
+	if c.VersionMetadata != "" {
+		version = fmt.Sprintf("%s+%s", version, c.VersionMetadata)
+	}
+
+	return version
+}
+
+func (c *VersionInfo) FullVersionNumber(rev bool) string {
+	var versionString bytes.Buffer
+
+	fmt.Fprintf(&versionString, "Nomad v%s", c.Version)
```

Comment:
dropped a "Pack" here

```suggestion
	fmt.Fprintf(&versionString, "Nomad Pack v%s", c.Version)
```

looks to be the cause of [this test failure](https://github.com/hashicorp/nomad-pack/actions/runs/6458991151/job/17533963690?pr=441#step:3:369) too.

Reply:
that ^ will re-break "fix tests" b9ffbe184f83dc15b4608764ade6801089d01d5c I think, but I figure we do want "Nomad Pack" in the output here:

```
$ ./bin/nomad-pack version
Nomad v0.0.1-techpreview.4
BuildDate ...etc...
```